src/peep/peep.c

```diff
@@ -7772,13 +7772,370 @@ static void peep_on_enter_or_exit_ride(rct_peep *peep, int rideIndex, int flags)
+	// FIXME: Is there a call to have the peep buy an item?
+	if (shopItem >= 32)
+		peep->item_extra_flags |= (1 << shopItem2);
+	else
+		peep->item_standard_flags |= (1 << shopItem2);
+
+/// loc_69B2C8:
+	if (shopItem == SHOP_ITEM_TSHIRT)
+		peep->tshirt_colour = ride->track_colour_main[0];
+
+/// loc_69B2E0:
+	if (shopItem == SHOP_ITEM_HAT)
+		peep->hat_colour = ride->track_colour_main[0];
+
+/// loc_69B2FB:
+	if (shopItem == SHOP_ITEM_BALLOON)
+		peep->balloon_colour = ride->track_colour_main[0];
+
+/// loc_69B316:
+	if (shopItem == SHOP_ITEM_UMBRELLA)
+		peep->umbrella_colour = ride->track_colour_main[0];
+
+/// loc_69B331:
+	if (shopItem == SHOP_ITEM_MAP)
+		sub_69A98C(peep);
+
+/// loc_69B33A:
+	uint16 dl;
+	if (shopItem >= 32)
+		dl = RCT2_ADDRESS(0x982310, uint8)[shopItem2];
+	else
+		dl = RCT2_ADDRESS(0x9822F4, uint8)[shopItem2];
+/// loc_69B34C:
+	peep->var_42 = min((peep->var_42 + dl), 255);
+
+/// loc_69B355:
+	if (shopItem == SHOP_ITEM_PHOTO)
+		peep->photo1_ride_ref = rideIndex;
+
+/// loc_69B35F:
+	if (shopItem == SHOP_ITEM_PHOTO2)
+		peep->photo2_ride_ref = rideIndex;
+
+/// loc_69B366:
+	if (shopItem == SHOP_ITEM_PHOTO3)
+		peep->photo3_ride_ref = rideIndex;
+
+/// loc_69B36D:
+	if (shopItem == SHOP_ITEM_PHOTO4)
+		peep->photo4_ride_ref = rideIndex;
+
+/// loc_69B374:
+	peep->window_invalidate_flags |= PEEP_INVALIDATE_PEEP_INVENTORY;
+	peep_update_sprite_type(peep);
+	uint32 ebx; // Looks like this is used to store the cost-per-item in order to calculate the net profit.
+	if (peep->flags & PEEP_FLAGS_TRACKING) {
+		RCT2_GLOBAL(0x13CE952,uint16) = peep->name_string_idx;
+		RCT2_GLOBAL((0x13CE952 + 2), uint32) = peep->id;
+		uint16 item_string = STR_SHOP_ITEM_INDEFINITE_BALLOON + shopItem; // bx
+		if (item_string >= 0x81C) // FIXME: Make sure this comparison is correct
+			item_string += 0x5A;
+		// loc_69B3B5:
+		RCT2_GLOBAL((0x13CE956 + 2), uint16) = item_string;
+		news_item_add_to_queue(2, STR_PEEP_TRACKING_NOTIFICATION_BOUGHT_X, peep->sprite_index);
+	}
+
+/// loc_69B3CC:
+	if (shopItem >= 32) {
+		/// loc_69B455:
+		if (0x00118F48 & (1 << shopItem2))
+			peep->no_of_food++;
+		/// loc_69B465:
+		if (0x00007030 & (1 << shopItem2))
+			peep->no_of_drinks++;
+		/// loc 69B475:
+		if (0x00000087 & (1 << shopItem2))
+			peep->no_of_souvenirs++;
+		/// loc_69B485:
+		// According to the doc on peep_spend_money(), 0x00F1AEC0
+		// is a global var for the "spend type".
+		RCT2_GLOBAL(0x00F1AEC0, uint32) = 0xEA;
+		// Another value that is known, but I don't know
+		// where it's used in the original code.
+		RCT2_GLOBAL(0x0141F56C, uint8) = 0x1C;
+		if (0x00118F48 & (1 << shopItem2)) {
+			RCT2_GLOBAL(0x00F1AEC0, uint32) = 0xE8;
+			RCT2_GLOBAL(0x0141F56C, uint8) = 0x24;
+		}
+		/// loc_69B4B1:
+		if (0x00007030 & (1 << shopItem2)) {
+			RCT2_GLOBAL(0x00F1AEC0, uint32) = 0x46;
+			RCT2_GLOBAL(0x0141F56C, uint8) = 0x24;
+		}
+		/// loc_69B4CC:
+		ebx = RCT2_ADDRESS(0x00982244, uint16)[shopItem2 * 4]; // TODO: Check stride
```

Comment:
This (and the similar one below) is an array of `money16` for storing the item cost (for example: umbrellas cost $2.00, so the array stores 20 or 0x14). It seems to be that they're both addresses to the same array, *but* the first part of the array for standard items is only 27 entries long, so the index of the first extra item would actually be 28. There's three other values, but I haven't divined their use yet.